File: algebra/arithmetic_progression/python/arithmetic_progression.py
'''nth term of Arithmetic Progression is a + (n-1)*d
sum of arithmetic progression is (n*(2*a + (n-1)*d))/2
where n=number of terms
a = first term of an AP
d= common difference between 2 consecutive terms of an AP'''

class arithmetic_progression:
    '''It will have 2 functions to calculate nth term as well as sum of an arithmetic progression'''
    def __init__(self,a,d,n):
        self.a=a
        self.d=d
        self.n=n
        
    def nthterm(self):
        return self.a + (self.n-1)*self.d #To return nth term of AP
    
    def sum_ap(self):
            return self.n*(self.a*2 + (self.n-1)*self.d)/2

#demo        
ap=arithmetic_progression(2,2,5)
print (ap.nthterm())
print (ap.sum_ap())        
        

File: algebra/arithmetic_progression/cpp/arithmetic_progression.cpp
#include <iostream>
using namespace std;

int main()
{
    int a, d, n, i, tn;
    int sum = 0;

    cout<<"Enter the first term value of the A.P. series: ";
    cin>>a;
    cout<<"Enter the total numbers in the A.P. series: ";
    cin>>n;
    cout<<"Enter the common difference of A.P. series: ";
    cin>>d;
    sum = (n * (2 * a + (n - 1)* d ))/ 2;
    tn = a + (n - 1) * d;
    cout<<"Sum of the A.P series is: ";
    for (i = a; i <= tn; i = i + d )
    {
        if (i != tn)
            cout<<i<<" + ";
        else
            cout<<i<<" = "<<sum;
    }
    return 0;
}
